pkg/registry/core/pod/storage/eviction.go

```diff
@@ -226,10 +226,33 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdb := &pdbs[0]
 		pdbName = pdb.Name
 
-		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
-			updateDeletionOptions = true
-			return nil
+		switch {
+		// If the pod healthy policy is not enabled, or no pod healthy policy is specified,
+		// just continue to use the original default behavior.
+		case !feature.DefaultFeatureGate.Enabled(features.PDBPodHealthyPolicy) || pdb.Spec.PodHealthyPolicy == "":
+			// If the pod is not ready, it doesn't count towards currentHealthy and desiredHealthy and we should not decrement disruptionsAllowed,
+			// then application guarded by the PDB is not disrupted at the moment and deleting unready pod will not disrupt it
+			if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
+				// make sure the pod status hasn't changed status before we delete
+				updateDeletionOptions = true
+				return nil
+			}
+		case pdb.Spec.PodHealthyPolicy == policyv1.PodReady:
```

Comment:
`PodReady` is identical with empty `""` policy, so I'd expect the condition be identical or squash this condition with above, right?

Reply:
That is the misconception that can arise... Please see https://github.com/kubernetes/kubernetes/pull/113375#discussion_r1006321318 and the comments here.

The difference is in `&& pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0` as this guards the application until the `DesiredHealthy` number is reached. That is Running pods have a chance to become ready.